polkadot/runtime/parachains/src/lib.rs

```diff
@@ -58,9 +58,9 @@ use primitives::{HeadData, Id as ParaId, ValidationCode};
 use sp_runtime::{DispatchResult, FixedU128};
 
 /// Trait for tracking message delivery fees on a transport protocol.
-pub trait FeeTracker {
+pub trait FeeTracker<Id = ()> {
```

Comment:
Why is this a type parameter and not an associated type?

Reply:
To be able to use a default and because I want to specify it without needing to use trait bounds

Reply:
Thinking about it more, it does make more sense to make it an associated type, but in that case I'd need to either have two types, something like `ExponentialPricePara` and `ExponentialPriceParent` or unify them both in a single associated type that works, like and enum with two variants `enum Destination { Para(ParaId), Parent }` or a `MultiLocation`.
What do you think?

Reply:
Thinking about it more now, given that I only ever have one implementation of this trait on any particular concrete type, I should make it an associated type instead

Reply:
It's an associated type now :)